src/vocab.rs

```diff
@@ -35,22 +35,27 @@ impl PyVocab {
         })
     }
 
-    fn ngram_indices(&self, word: &str) -> PyResult<Option<Vec<(String, usize)>>> {
+    #[allow(clippy::type_complexity)]
+    fn ngram_indices(&self, word: &str) -> PyResult<Option<Vec<(String, Option<usize>)>>> {
```

Comment:
Oh, I didn't realize that the tuple became more complex. Maybe you could add a type alias

```
type NGramIndex = (String, Option<usize>);
```

?

Reply:
Not too big of a fan of using aliases like this because I usually end up looking for its definition anyways. Tho, I pushed the change.